tests/MongoDB.Driver.Core.Tests/Core/Compression/CompressorsTests.cs

```diff
@@ -30,6 +32,91 @@ public class CompressorsTests
     {
         private static string __testMessage = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz";
 
+        [Theory]
+        [InlineData(CompressorType.Snappy)]
+        public void Compressor_should_read_the_previously_written_big_message_or_throw_the_exception_if_the_current_platform_is_not_supported(CompressorType compressorType)
+        {
+            var message = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz";
+
+            var compressedMessage = "158;2;100;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;254;26;0;254;26;0;254;26;0;254;26;0;1;26";
```

Comment:
I've checked which bytes are expected by a server via logic based on `SnappyCodec` (described in the PR description).
So, `compressedMessage`, `decompressedMessage` contains correct bytes which server expects as a request and returns as a response.